### PR TITLE
#166085534 deleting updating article 166085534

### PR DESCRIPTION
**Pull request by felixkiryowa**
#### What does this PR do?

- Deleting and Updating Articles

#### Description of Task to be completed?
-  A user should be able to delete articles and edit articles only their articles when they have logged in.

#### How should this be manually tested?

- Fetch this branch `git fetch origin  ft-deleting-updating-article-166085534 `
- Checkout to the branch `git checkout   ft-deleting-updating-article-166085534 `
- Run `npm install`
- Run `npm run start:dev`
- Login into the application
- create article
- Got to dashboard and click  `Read More ...` to view the article in detail 
- Scroll down and click on `edit` or `delete` button 

#### What are the relevant pivotal tracker stories?
[ #166085534](https://www.pivotaltracker.com/story/show/166085534)
Screenshot 2019-05-20 at 16 56 53
Screenshot 2019-05-20 at 16 59 40
Screenshot 2019-05-20 at 16 59 50

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,10 +55,11 @@
     "react-toolbox": "^2.0.0-beta.13",
     "redux": "^4.0.1",
     "redux-devtools-extension": "^2.13.8",
-    "redux-mock-store": "^1.5.3",
     "redux-logger": "^3.0.6",
+    "redux-mock-store": "^1.5.3",
     "redux-thunk": "^2.3.0",
     "sass-loader": "^7.1.0",
+    "sinon": "^7.3.2",
```
Comment (dannylwe) [inline package.json]: @felixkiryowa what is this package for?

File: src/actions/articlesActions.js [hunk 2]
```diff
@@ -7,11 +7,16 @@ import {
...
+  DELETE_AN_ARTICLE_FAILURE,
+  EDIT_AN_ARTICLE_SUCCESS,
+  EDIT_AN_ARTICLE_FAILURE,
+
 } from './types';
 
 const postArticleFailure = err => ({
   type: POST_ARTICLE_FAILURE,
-  payload: err.response.data.errors,
+  payload: 'Errors have occured',
```
Comment (dannylwe) [inline src/actions/articlesActions.js]: @felixkiryowa kindly have descriptive error messages

File: src/actions/articlesActions.js [hunk 3]
```diff
@@ -21,46 +26,92 @@ const postArticleSuccess = response => ({
...
-export const getSingleArticle = slug => (dispatch) => {
-  axios.get(`${baseURL}/articles/${slug}`)
-    .then((response) => {
-      dispatch(fetchAnArticleSuccess(response));
-    }).catch((err) => {
-      dispatch(fetchAnArticleFailure(err.response.data));
-      notify.show('Errors have occured', 'error', 4000);
+
+
+const baseURL = 'https://ah-backend.herokuapp.com/api';
```
Comment (dannylwe) [inline src/actions/articlesActions.js:66]: @felixkiryowa kindly use the axios intance in src

File: src/pages/Routes.jsx [hunk 4]
```diff
@@ -30,6 +32,7 @@ class Routes extends Component {
           <Switch>
             <Route exact path="/" component={LandingComponent} />
             <Route path="/register" component={SignupComponent} />
+            <Route path="/articles/edit/:slug" component={EditArticleComponent} />
```
Comment (dannylwe) [inline src/pages/Routes.jsx]: @felixkiryowa your routes should be protected as private routes

File: src/pages/singleArticle/index.jsx [hunk 5]
```diff
@@ -11,10 +15,17 @@ export class SingleArticles extends React.Component {
     getArticle(slug);
   }
 
+  deleteAnArticle = (slug) => {
+    // const { history } = this.props;
```
Comment (dannylwe) [inline src/pages/singleArticle/index.jsx]: @felixkiryowa do you need this commented code?

File: src/pages/singleArticle/tests/__snapshots__/editArticle.test.jsx.snap [hunk 6]
```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Edit Article Page Component Should render without crashing 1`] = `ReactWrapper {}`;
```
Comment (dannylwe) [inline src/pages/singleArticle/tests/__snapshots__/editArticle.test.jsx.snap:3]: @felixkiryowa this is not what a snapshot is supposed to look like. You can use react-test-renderer

**Review by dannylwe — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (felixkiryowa) [reply package.json]: Worked on this feedback

File: src/components/Articles/editArticle.jsx [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+  },
+
+) => (
+  <div className="container">
+    <br />
+    <br />
+    <br />
+    <div className="row">
+      <div className="col-md-1" />
+      <div className="col-md-10" style={{ paddingLeft: '20px' }}>
```
Comment (ezrogha) [inline src/components/Articles/editArticle.jsx]: Kindly remove these inline styles

File: src/components/Articles/editArticle.jsx [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+        <h4 className="article-heading">Update Article</h4>
+        <form onSubmit={onSubmit}>
+          <div className="form-group">
+            <label
+              id="article-title"
+              htmlFor="title"
+              className="control-label col-md-12 col-sm-12 col-xs-12"
+              style={{ color: '#8D8D8D' }}
```
Comment (ezrogha) [inline src/components/Articles/editArticle.jsx]: Remove these inline styles

File: src/components/Articles/editArticle.jsx [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+              <br />
+              <input
+                type="text"
+                className="form-control"
+                id="title"
+                name="title"
+                placeholder="Enter Article Title"
+                value={title}
+                onChange={handleChange}
+                style={{ height: '50px', backgroundColor: '#F3F3F3' }}
```
Comment (ezrogha) [inline src/components/Articles/editArticle.jsx]: Remove this inline style

File: src/components/Articles/editArticle.jsx [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+              />
+            </label>
+            <br />
+            <br />
+            <br />
+            <label
+              id="article-description"
+              htmlFor="description"
+              className="control-label col-md-12 col-sm-12 col-xs-12"
+              style={{ color: '#8D8D8D' }}
```
Comment (ezrogha) [inline src/components/Articles/editArticle.jsx]: Remove this inline style

File: src/components/Articles/editArticle.jsx [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+              <br />
+              <input
+                type="text"
+                className="form-control"
+                id="description"
+                name="description"
+                placeholder="Enter Article Description"
+                value={description}
+                onChange={handleChange}
+                style={{ height: '80px', backgroundColor: '#F3F3F3' }}
```
Comment (ezrogha) [inline src/components/Articles/editArticle.jsx]: remove this inline style

File: src/components/Articles/editArticle.jsx [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+              />
+            </label>
+            <br />
+            <br />
+            <br />
+            <label
+              id="article-body"
+              htmlFor="body"
+              className="control-label col-md-12 col-sm-12 col-xs-12"
+              style={{ color: '#8D8D8D' }}
```
Comment (ezrogha) [inline src/components/Articles/editArticle.jsx]: remove this inline style

File: src/components/Articles/editArticle.jsx [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+              <textarea
+                type="text"
+                className="form-control"
+                id="body"
+                name="body"
+                aria-describedby="article"
+                placeholder="Enter Article Body"
+                value={body}
+                onChange={handleChange}
+                style={{ height: '200px', backgroundColor: '#F3F3F3' }}
```
Comment (ezrogha) [inline src/components/Articles/editArticle.jsx]: Remove this inline style

File: src/tests/actions/Articles.test.js [hunk 14]
```diff
@@ -0,0 +1,177 @@
+import moxios from 'moxios';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+// import axiosInstance from '../../helpers/axiosInstance';
```
Comment (ezrogha) [inline src/tests/actions/Articles.test.js]: Remove this comment

File: src/tests/actions/Articles.test.js [hunk 15]
```diff
@@ -0,0 +1,177 @@
...
+    }];
+    const slug = 'new-of-testing-16';
+    moxios.wait(() => {
+      const request = moxios.requests.mostRecent();
+      request.respondWith({
+        status: 200,
+        response: expectedResponse,
+      });
+    });
+    // configure Mock store
```
Comment (ezrogha) [inline src/tests/actions/Articles.test.js]: Remove this comment

**Review by ezrogha — CHANGES_REQUESTED** [review]
Good work @felixkiryowa,
Kindly work on this feedback

File: src/actions/articlesActions.js (hunk 3 above)
Comment (felixkiryowa) [reply src/actions/articlesActions.js:66]: This baseURL is being used for those endpoints that do not need user authorization like fetching articles and getting a single single

**Review by ezrogha — APPROVED** [review]
well done with the feedback @felixkiryowa,
LGTM

File: src/components/Articles/article.scss [hunk 16]
```diff
@@ -171,6 +171,51 @@ div.ReactTags__suggestions ul li.ReactTags__activeSuggestion {
...
+.input-labels {
+    color: '#8D8D8D'
+}
+
+::placeholder {
+  color: #6c757d;
+}
+
+.div-holder {
+    paddingLeft: '20px'
```
Comment (dannylwe) [inline src/components/Articles/article.scss]: @felixkiryowa kindly change to px to rem and "padding-left"

File: src/tests/actions/ArticlesActions.test.js [hunk 17]
```diff
@@ -1,111 +1,196 @@
...
+      payload: 'Errors have occured',
+    }];
+    moxios.wait(() => {
+      const request = moxios.requests.mostRecent();
+      request.respondWith({
+        status: 404,
+        response: expectedState,
+      });
+    });
+    // configure Mock store
```
Comment (dannylwe) [inline src/tests/actions/ArticlesActions.test.js]: @felixkiryowa kindly remove comments

File: src/tests/actions/ArticlesActions.test.js [hunk 18]
```diff
@@ -118,39 +203,91 @@ describe('Testing Articles', () => {
...
+      request.respondWith({
+        status: 200,
+        response: expectedResponse,
+      });
     });
+    // configure Mock store
     const store = mockStore({});
+    const historyMock = { history: { push: jest.fn() } };
+    const slug = 'new-article';
+    // call the getBucketLists async action creator
```
Comment (dannylwe) [inline src/tests/actions/ArticlesActions.test.js]: @felixkiryowa kindly remove comments

**Review by dannylwe — CHANGES_REQUESTED** [review]

**Review by dannylwe — CHANGES_REQUESTED** [review]
@felixkiryowa you have two commits, kindly squash

**Review by dannylwe — APPROVED** [review]

**Review by ezrogha — APPROVED** [review]
LGTM

**Review by hadijahkyampeire — APPROVED** [review]
Well done

**Pull request merged**